routes/dao_coin_exchange.go

```diff
@@ -0,0 +1,183 @@
+	if err := decoder.Decode(&requestData); err != nil {
+		_AddBadRequestError(
+			ww,
+			fmt.Sprintf("GetDAOCoinLimitOrders: Problem parsing request body: %v", err),
+		)
+		return
+	}
+
+	if requestData.DAOCoin1CreatorPublicKeyBase58CheckOrUsername == "" &&
+		requestData.DAOCoin2CreatorPublicKeyBase58CheckOrUsername == "" {
+		_AddBadRequestError(
+			ww,
+			fmt.Sprint("GetDAOCoinLimitOrders: Must provide either a "+
+				"DAOCoin1CreatorPublicKeyBase58CheckOrUsername or DAOCoin2CreatorPublicKeyBase58CheckOrUsername "+
+				"or both"),
+		)
+		return
+	}
+
+	utxoView, err := fes.backendServer.GetMempool().GetAugmentedUniversalView()
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("GetDAOCoinLimitOrders: Problem fetching utxoView: %v", err))
+		return
+	}
+
+	coin1PKID := &lib.ZeroPKID
+	coin2PKID := &lib.ZeroPKID
+
+	coin1ProfilePublicKeyBytes := lib.ZeroPublicKey.ToBytes()
+	coin2ProfilePublicKeyBytes := lib.ZeroPublicKey.ToBytes()
+
+	coin1ProfilePublicBase58Check := ""
+	coin2ProfilePublicBase58Check := ""
+
+	if requestData.DAOCoin1CreatorPublicKeyBase58CheckOrUsername != "" {
+		coin1ProfilePublicKeyBytes, _, err = fes.GetPubKeAndProfileEntryForUsernameOrPublicKeyBase58Check(
+			requestData.DAOCoin1CreatorPublicKeyBase58CheckOrUsername,
+			utxoView,
+		)
+		if err != nil {
+			_AddBadRequestError(
+				ww,
+				fmt.Sprint("GetDAOCoinLimitOrders: Invalid DAOCoin1CreatorPublicKeyBase58CheckOrUsername"),
+			)
+			return
+		}
+		coin1PKID = utxoView.GetPKIDForPublicKey(coin1ProfilePublicKeyBytes).PKID
+		coin1ProfilePublicBase58Check = lib.Base58CheckEncode(coin1ProfilePublicKeyBytes, false, fes.Params)
+	}
+
+	if requestData.DAOCoin2CreatorPublicKeyBase58CheckOrUsername != "" {
+		coin2ProfilePublicKeyBytes, _, err = fes.GetPubKeAndProfileEntryForUsernameOrPublicKeyBase58Check(
+			requestData.DAOCoin2CreatorPublicKeyBase58CheckOrUsername,
+			utxoView,
+		)
+		if err != nil {
+			_AddBadRequestError(
+				ww,
+				fmt.Sprint("GetDAOCoinLimitOrders: Invalid DAOCoin2CreatorPublicKeyBase58CheckOrUsername"),
+			)
+			return
+		}
+		coin2PKID = utxoView.GetPKIDForPublicKey(coin2ProfilePublicKeyBytes).PKID
+		coin2ProfilePublicBase58Check = lib.Base58CheckEncode(coin2ProfilePublicKeyBytes, false, fes.Params)
+	}
+
+	adapter := utxoView.GetDbAdapter()
+
+	ordersBuyingCoin1, err := adapter.GetAllDAOCoinLimitOrdersForThisDAOCoinPair(coin1PKID, coin2PKID)
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("GetDAOCoinLimitOrders: Error getting limit orders: %v", err))
+		return
+	}
+
+	ordersSellingCoin1, err := adapter.GetAllDAOCoinLimitOrdersForThisDAOCoinPair(coin2PKID, coin1PKID)
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("GetDAOCoinLimitOrders: Error getting limit orders: %v", err))
+		return
+	}
+
+	var response []DAOCoinLimitOrder
+
+	for _, order := range ordersBuyingCoin1 {
+		transactorPublicKey := utxoView.GetPublicKeyForPKID(order.TransactorPKID)
+		response = append(response, DAOCoinLimitOrder{
```

Comment:
it might be better to write a function `_daoCoinLimitOrderToResponse` to handle this.